.cbmc-batch/include/proof_helpers/make_common_data_structures.h

```diff
@@ -16,21 +16,63 @@
 #pragma once
 
 #include <aws/common/array_list.h>
+#include <aws/common/byte_buf.h>
 #include <aws/common/common.h>
-
-#include "nondet.h"
+#include <aws/common/string.h>
+#include <proof_helpers/nondet.h>
+#include <proof_helpers/proof_allocators.h>
 
 #include <stdlib.h>
 
-#define MAX_INITIAL_ITEM_ALLOCATION 2
+#define MAX_INITIAL_ITEM_ALLOCATION 5
 #define MAX_ITEM_SIZE 2
-#define MAX_STR_LEN 32
-#define MAX_BUF_LEN 32
+#define MAX_STR_LEN 2
+#define MAX_BUF_LEN 2
 
 #define ASSUME_VALID_MEMORY(ptr) ptr = malloc(sizeof(*(ptr)))
 #define ASSUME_VALID_MEMORY_COUNT(ptr, count) ptr = malloc(sizeof(*(ptr)) * (count))
 #define ASSUME_DEFAULT_ALLOCATOR(allocator) allocator = aws_default_allocator()
-#define ASSUME_ARBITRARY_ARRAY_LIST(list, item_count, item_size)                                                       \
-    list = make_arbitrary_array_list((item_count), (item_size))
+#define ASSUME_CAN_FAIL_ALLOCATOR(allocator) allocator = can_fail_allocator()
+#define ASSUME_ARBITRARY_ARRAY_LIST(list, initial_item_allocation, item_size)                                          \
+    list = make_arbitrary_array_list((initial_item_allocation), (item_size))
+#define ASSUME_NONDET_ARRAY_LIST(list) list = make_nondet_array_list()
+#define ASSUME_BOUNDED_ARRAY_LIST(list, max_initial_item_allocation, max_item_size)                                    \
+    list = make_bounded_array_list((max_initial_item_allocation), (max_item_size))
+
+struct aws_array_list *make_arbitrary_array_list(size_t initial_item_allocation, size_t item_size);
+
+struct aws_array_list *make_nondet_array_list();
+
+struct aws_array_list *make_bounded_array_list(size_t max_initial_item_allocation, size_t max_item_size);
+
+int compare(const void *a, const void *b);
+
+/**
+ * Makes a byte_buf, with as much nondet as possible, len < max, valid backing storage
+ */
+struct aws_byte_cursor make_arbitrary_byte_cursor_nondet_len_max(size_t max);
```

Comment:
It would be nice to have a line of explanation about the other helper functions, too.

Reply:
Done